### PR TITLE
Form association

### PR DESCRIPTION
**Pull request by annevk**
For https://github.com/whatwg/html/pull/2874.

### CODE REVIEW

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/259326710)

# Firefox (nightly) #

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/259326711)

# Sauce (safari) #

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/259326724)

# Chrome (unstable) #

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/259326726)

# Sauce (MicrosoftEdge) #

File: html/semantics/forms/form-control-infrastructure/association.window.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+test(() => {
+  const form = document.createElement("form"),
+        input = document.createElement("input");
+
+  form.appendChild(input);
+  assert_equals(input.form === form);
```
Comment (tkent-google) [inline html/semantics/forms/form-control-infrastructure/association.window.js]: should be
  assert_equals(input.form, form);

File: html/semantics/forms/form-control-infrastructure/form_owner_and_table_3.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+<script>
+test(() => {
+  const input = document.querySelector("input"),
+        form = document.querySelector("form");
+  assert_equals(input.form, form);
+  document.getElementById("2").appendChild(form.parentNode);
+  assert_equals(input.form, form);
+  document.getElementById("2").appendChild(input);
+  assert_equals(form.input, null);
+}, "parser inserted flag is only reset sometimes");
```
Comment (tkent-google) [inline html/semantics/forms/form-control-infrastructure/form_owner_and_table_3.html]: 'sometime' is unclear.  How about:
 "parser inserted flag is not reset by insertions with the owner form, but reset by by removal from the owner form"

**Review by tkent-google — APPROVED** [review]

**Comment by annevk** [comment]
Commit message:
```
HTML: form association

For https://github.com/whatwg/html/pull/2874.
```

**Pull request merged**